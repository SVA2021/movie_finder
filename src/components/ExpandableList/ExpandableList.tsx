import {Box, Grid, Typography} from '@mui/material';
import {FC} from 'react';
import {SmallCard} from '../SmallCard/SmallCard';
import {StyledButton} from '../Styled_Button/Styled_Button';

interface ExpandableListProps {
    title: string
    data: any[]
    page: number
    totalPages: number
    addPage: () => void
}

export const ExpandableList: FC<ExpandableListProps> = ({title, data, page, totalPages, addPage}) => {
    return (
        <Box pt={1}>
            <Typography
                ml={{lg: 15, xl: 20}}
                gutterBottom
                variant="h3"
                component={'h2'}
                textTransform={'capitalize'}
            >
                {title}
            </Typography>
            <Grid container>
                {
                    data.map((item, index) =>
                        <Grid key={index} item>
                            <SmallCard item={item} />
                        </Grid>
                    )
                }
            </Grid>
            <StyledButton onClick={addPage} disabled={page === totalPages} >Load more</StyledButton>
        </Box>
    );
};