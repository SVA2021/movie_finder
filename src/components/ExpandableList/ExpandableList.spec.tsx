import {render, screen, fireEvent} from "@testing-library/react";
import {ExpandableList} from "./ExpandableList";
import {BrowserRouter} from 'react-router-dom';

const FakeCard = {
  "id": 325381,
  "nameRu": "fakeNameRu",
  "nameEn": "fakeNameEng",
  "year": "2000",
  "filmLength": "10:10",
  "rating": "7.0",
  "posterUrlPreview": "fakePosterUrlPreview",
}

describe('render ExpandableList', () => {

  const onClick = jest.fn();

  const setUp = () => render(
    <BrowserRouter>
      <ExpandableList
        title={"fakeTitle"}
        data={[FakeCard]}
        page={1}
        totalPages={3}
        addPage={onClick}
      />
    </BrowserRouter>
  );

  const disabledButtonSetUp = () => render(
    <BrowserRouter>
      <ExpandableList
        title={""}
        data={[FakeCard]}
        page={1}
        totalPages={1}
        addPage={onClick}
      />
    </BrowserRouter>
  );

  const emptySetUp = () => render(
    <BrowserRouter>
      <ExpandableList
        title={""}
        data={[]}
        page={0}
        totalPages={0}
        addPage={onClick}
      />
    </BrowserRouter>
  );

  it('snapshot test', () => {
    setUp();
    expect(screen.getByTestId('expandable-list')).toMatchSnapshot();
  });

  it('check title', () => {
    setUp();
    expect(screen.getByText(/fakeTitle/i)).toBeInTheDocument();
  });

  it('check button', () => {
    setUp();
    fireEvent.click(screen.getByText('Еще'))
    expect(onClick).toBeCalled();
  });

  it('check button is disabled', () => {
    disabledButtonSetUp();
    expect(screen.getByText('Еще')).toBeDisabled();
  });

  it('check empty data placeholder', () => {
    emptySetUp();
    expect(screen.getByText(/нет данных/i)).toBeInTheDocument();
  });

})