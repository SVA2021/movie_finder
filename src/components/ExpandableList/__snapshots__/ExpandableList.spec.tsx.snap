// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ExpandableList snapshot test 1`] = `
<div
  class="MuiBox-root css-zoser8"
  data-testid="expandable-list"
>
  <h2
    class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom css-9q8iy8-MuiTypography-root"
  >
    fakeTitle
  </h2>
  <div
    class="MuiGrid-root MuiGrid-container css-l4fv9m-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 MuiGrid-grid-lg-2 css-57ri8o-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-4ob670-MuiPaper-root-MuiCard-root"
        data-testid="smallCard"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root css-tqw9zv-MuiButtonBase-root-MuiCardActionArea-root"
          tabindex="0"
          type="button"
        >
          <img
            alt="fakeNameRu"
            class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
            src="fakePosterUrlPreview"
            width="100%"
          />
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-1ydyjjr-MuiTypography-root"
            >
              fakeNameRu
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-12e029l-MuiTypography-root"
            >
              fakeNameEng
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
            >
              2000
              ,  
              10ч10м
            </p>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <p
          class="MuiTypography-root MuiTypography-h3 css-1uovp3e-MuiTypography-root"
          data-testid="rating-badge"
        >
          7.0
        </p>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root css-4sywta"
  >
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Еще
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
